[CI] TermsAggregatorTests.testSupportedFieldTypes fails reproducibly

This looks to be a newly added test which has failed 3 times already. This also reproduces locally for me.

https://gradle-enterprise.elastic.co/s/xqoxiluwuzono/tests/nfxodyych4uas-al6l6qgxmtvfa?open-stacktraces=WzBd

```
java.lang.AssertionError: Aggregator [terms] supports field type [ip] but executing against the field threw an exception: [class org.apache.lucene.index.ParallelCompositeReader cannot be cast to class org.apache.lucene.index.DirectoryReader (org.apache.lucene.index.ParallelCompositeReader and org.apache.lucene.index.DirectoryReader are in unnamed module of loader 'app')]Close stacktrace
at __randomizedtesting.SeedInfo.seed([37F644DC7A7897F3:90E5F9E8055C4EC0]:0)
at org.junit.Assert.fail(Assert.java:88)
at org.elasticsearch.search.aggregations.AggregatorTestCase.testSupportedFieldTypes(AggregatorTestCase.java:661)
```

--- a/test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java
@@ -605,7 +605,6 @@ public abstract class AggregatorTestCase extends ESTestCase {
      *
      * Exception types/messages are not currently checked, just presence/absence of an exception.
      */
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/52681")
     public void testSupportedFieldTypes() throws IOException {
         MapperRegistry mapperRegistry = new IndicesModule(Collections.emptyList()).getMapperRegistry();
         Settings settings = Settings.builder().put("index.version.created", Version.CURRENT.id).build();
@@ -647,7 +646,7 @@ public abstract class AggregatorTestCase extends ESTestCase {
                 indexWriter.close();
 
                 try (IndexReader indexReader = DirectoryReader.open(directory)) {
-                    IndexSearcher indexSearcher = newSearcher(indexReader, true, true);
+                    IndexSearcher indexSearcher = newIndexSearcher(indexReader);
                     AggregationBuilder aggregationBuilder = createAggBuilderForTypeTest(fieldType, fieldName);
 
                     // TODO in the future we can make this more explicit with expectThrows(), when the exceptions are standardized

